WordPress/gutenberg
# Remove base url from link control search results

## What?


Removes the site's `baseURL` (if available) from the Link Control search results:

Closes https://github.com/WordPress/gutenberg/issues/52572

## Why?


Polish to limit UI noise. See Issue.

## How?


Display only the _path_ of the search result URL if the type is anything other than a manually entered URL.

Props to @Mamaduka for this suggested approach.


## Testing Instructions





- Repeat the below in Post, Widget and Site editors
- Make sure you have some Posts and Pages
- Open Editor
- Add text block.
- Create a link and search for one of the Pages/Posts
- See results with the site's base URL stripped off (i.e. `localhost:8888/sample-page` becomes `/sample-page/`).
- Verify that manually entering a URL such as `https://wordpress.org/hello-world` does _not_ strip the site URL.

### Testing Instructions for Keyboard


## Screenshots or screencast 

<img width="457" alt="Screenshot 2023-09-18 at 14 13 49" src="https://github.com/WordPress/gutenberg/assets/444434/45536bd6-beb9-4f68-bf9d-e1b19ad46c01">
